packages/circuits/semaphore.circom

```diff
@@ -4,22 +4,64 @@ include "babyjub.circom";
 include "poseidon.circom";
 include "binary-merkle-root.circom";
 
+// The Semaphore circuit can be divided into 3 main parts.
+// The first part involves the generation of the Semaphore identity,
+// i.e. the public key and its hash, which is called the commitment
+// and is used as a public value.
+// In the second part, it is verified whether the identity commitment is part
+// of the Merkle tree, i.e. the Semaphore group. That is, a proof of membership is verified.
+// The third part covers the generation of a nullifier, i.e. the hash of the scope of the proof
+// and the secret used to derive the public key (secret scalar). The nullifier is used to prevent the same
```

Comment:
Probably worth adding an high level representation:

// h(scope, secret) ...

Reply:
I think the high-level representation is the code itself at line `60`. These comments will not be part of TypeDoc, so they will still see the code.